test_browser/test_browser.ts

```diff
@@ -0,0 +1,34 @@
+import { serveFile } from "https://deno.land/std@0.172.0/http/file_server.ts"
+import { serve } from "https://deno.land/std@0.172.0/http/server.ts"
+import { assertEquals } from "https://deno.land/std@0.172.0/testing/asserts.ts"
+
+import { PORT, STATIC_DIR } from "./constants.ts"
+import EXPECTED_RESULTS from "./expected_results.ts"
+
+const handler = async (req: Request): Promise<Response> => {
+  const isPOST = req.method === "POST"
+  const pattern = new URLPattern({ pathname: "/:file_path" })
+  const filePath = pattern.exec(req.url)?.pathname.groups.file_path
+  const isFinished = filePath === "end_of_tests"
+
+  if (isFinished) Deno.exit(0)
```

Comment:
Not the ideal way to terminate the process. You can abort controller and supply its signal to `serve`'s props. Then call `abort` when necessary.

Reply:
Great! In the future, no need to add comments like `Thanks, applied!`. Just mark the comment as resolved.